index.js

```diff
@@ -29,33 +29,38 @@ exports.RaveLevel = class RaveLevel extends ManyLevelGuest {
     this[kOptions] = { keyEncoding, valueEncoding }
     this[kConnect] = this[kConnect].bind(this)
     this[kDestroy] = this[kDestroy].bind(this)
+  }
 
-    this.open(this[kConnect])
+  _open(options, cb) {
+    super._open(options, (err) => {
+      this[kConnect](err, cb);
+    });
   }
 
-  [kConnect] () {
-    // Check database status every step of the way
-    if (this.status !== 'open') {
-      return
+  [kConnect] (err, cb) {
+    if (err) {
+      return cb(err);
     }
```

Comment:
I believe the reason for the check against `this.status !== 'open'` is because the `kConnect` gets called repeatedly, in case of error, to retry, when server disconnects, etc.... So if a user calls a `db.close()`, it would get called again and would try to re-open it, so it checks the status.
I think that check needs to remain, but change into a check against `open` and `opening`.